virtiofs: adds non-privileged virtiofsd sharing (#734)
kubevirt/user-guide: docs/virtual_machines/disks_and_volumes.md

`Virtiofs` is a shared file system that lets VMs access a directory tree on the host.
Further details can be found at [Official Virtiofs Site](https://virtio-fs.gitlab.io/).

### Non-Privileged and Privileged Sharing Modes

KubeVirt supports two PVC sharing modes: non-privileged and privileged.

The **non-privileged mode** is enabled by default. This mode has the advantage of not requiring any

> administration privileges for creating the VM. However, it has some limitation:

Author: Done.

> - The virtiofsd daemon (i.e., the daemon in charge to share the PVC with the VM) will run with the QEMU UID/GID (107),

Reviewer: ```suggestion
- The virtiofsd daemon (the daemon in charge of sharing the PVC with the VM) will run with the QEMU UID/GID (107),
```

Author: Done.
